src/system/file_path.hh

```diff
@@ -76,11 +76,11 @@ public:
     /**
      * Set:
      * - working directory (used only if the output directory is relative)
-     * - root directory (of the main input file)
+     * - input root directory (path of the main input file)
      * - input directory to replace ${INPUT} place holder
      * - output directory used as prefix to the output files (relative output dirs are relative to the working directory)
      */
-    static void set_io_dirs(const string working_dir, const string root_input,const string input,const string output);
+    static string set_io_dirs(const string main_input_file, const string input, const string output, const string working_dir=".");
```

Comment:
Both output and input directory, if specified as relative should be relative to the root directory of the main_input_file.
Then we can possibly remove working directory completely. In order to simplify tests, where one may use several input files and setting non-existent main yaml file is awkward, I suggest to keep several set_io_* methods, see next comment.